linkerd/meshtls/Cargo.toml

```
@@ -9,6 +9,7 @@ publish = false
 [features]
 rustls = ["linkerd-meshtls-rustls", "__has_any_tls_impls"]
 boring = ["linkerd-meshtls-boring", "__has_any_tls_impls"]
+fips = ["linkerd-meshtls-boring/fips"]
```

this feature flag will always enable the optional dependency on `linkerd-meshtls-boring` if it's enabled, even if the `boring` feature flag is not enabled. is that the intended behavior?

we could use a `?` feature (see [here](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features) for details) to only have the `meshtls` crate's `fips` feature enable the `linkerd-meshtls-boring` crate's `fips` feature _if_ some other feature flag (e.g. this crate's `boring` feature) enables the `linkerd-meshtls-boring` dependency):
```suggestion
fips = ["?linkerd-meshtls-boring/fips"]
```
i believe that would be the intended behavior here.

alternatively, if we wanted to have the `fips` feature always enable the dependency on `linkerd-meshtls-boring`, we may as well just have it enable this crate's `boring` feature flag, so that it automatically enables BoringSSL. this would make the feature flag combinations a bit simpler, but (as I discussed in my other comment) it would preclude the possibility to make this crate's `fips` feature enable FIPS support for _any_ TLS implementation that's enabled, in the future...